src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -73,16 +75,17 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
     }
   }
 
-  private static String findRuntime(StageFlexibleConfiguration config) {
+  @VisibleForTesting
+  static String findRuntime(StageFlexibleConfiguration config) {
     // verification for app.yaml that contains runtime:java
     Path appYaml = config.getAppEngineDirectory().toPath().resolve(APP_YAML);
     String runtime = null;
     try {
       if (Files.isRegularFile(appYaml)) {
         runtime = new AppYaml(appYaml).getRuntime();
       }
-    } catch (IOException e) {
-      log.warning("Unable to determine runtime: error parsing app.yaml : " + e.getMessage());
+    } catch (IOException | ScannerException | ParserException ex) {
```

Comment:
I see.  It think it makes more sense to me for this method to throw  a checked InvalidAppYamlException that wraps the cause and lets the calling staging code log it and carry on.